src/test/predict_validation_test.cpp

```diff
@@ -2048,4 +2049,66 @@ TYPED_TEST(PredictValidationString1DTest, string_not_allowed_with_demultiplexer)
     EXPECT_EQ(status, ovms::StatusCode::NOT_IMPLEMENTED);
 }
 
+template <typename TensorType>
+class PredictValidationStringNativeTest : public ::testing::Test {
+protected:
+    TensorType request;
+    const char* tensorName = DUMMY_MODEL_INPUT_NAME;
+    ovms::tensor_map_t mockedInputsInfo;
+    void SetUp() override {
+        auto shape1d = ovms::Shape{-1};
+        mockedInputsInfo[tensorName] = std::make_shared<ovms::TensorInfo>(tensorName, ovms::Precision::STRING, shape1d, ovms::Layout{"N..."});
+    }
+};
+
+TYPED_TEST_SUITE(PredictValidationStringNativeTest, MyTypes);
+
+TYPED_TEST(PredictValidationStringNativeTest, positive) {
+    // bs=1
+    std::vector<std::string> inputStrings = {"String_123"};
+    prepareInferStringRequest(this->request, this->tensorName, inputStrings);
+    auto status = ovms::request_validation_utils::validate(this->request, this->mockedInputsInfo, "dummy", ovms::model_version_t{1});
+    EXPECT_EQ(status, ovms::StatusCode::OK);
+    // bs=2
+    this->request.Clear();
+    inputStrings = {"String_123", "other"};
+    prepareInferStringRequest(this->request, this->tensorName, inputStrings);
+    status = ovms::request_validation_utils::validate(this->request, this->mockedInputsInfo, "dummy", ovms::model_version_t{1});
+    EXPECT_EQ(status, ovms::StatusCode::OK);
+}
+
+TYPED_TEST(PredictValidationStringNativeTest, negative_wrong_request_shape) {
+    prepareInferStringInputWithTwoDimensionShapeTensor(this->request, this->tensorName);
+    auto status = ovms::request_validation_utils::validate(this->request, this->mockedInputsInfo, "dummy", ovms::model_version_t{1});
+    EXPECT_EQ(status, ovms::StatusCode::INVALID_NO_OF_SHAPE_DIMENSIONS);
+}
+
+TYPED_TEST(PredictValidationStringNativeTest, positive_over_1gb) {
+    std::string longString(1024 * 1024 * 512 * 1, 'a');                            // 512mb
```

Comment:
Should it be positive?

Reply:
There is nothing wrong with such payload, grpc server parameter may limit the message size at server level